### ISSUE TITLE
🐛 Mob Created by zone:insertDynamicEntity Crashes Windower Players Who Nuke It

### ISSUE THREAD

**Issue by Dev-Roland** [DerpyProjectGroup/topaz#2008]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)

## Branch affected by issue 

`base`

## Steps to reproduce

1. Load the **Timers** plugin
2. Execute code following this template to spawn a custom mob
```
local mob = zone:insertDynamicEntity({
                objtype = xi.objType.MOB,
                name = "Name Here",
                x = -6,
                y = -6,
                z = 6,
                rotation = 0,
                groupId = 69,
                groupZoneId = 666,

            onMobSpawn = function(mob)
            SetServerVariable("Variable_Here", 0)
            end,

            onMobDeath = function(mob, player, isKiller, noKiller)
                player:setCharVar("Variable_Here",1)
            end,
            releaseIdOnDeath = true,
})
```
3. On a Windower player, nuke the mob
4. Crash to desktop the moment the spell stops casting, before its animation plays

RELATED ISSUE:
With the Battlemod addon loaded these mobs also seem to output nothing to the player's chat. Apparently they aren't sending the expected damage/spell/etc chat packets.

## Expected behavior

The player does not crash to desktop when nuking the custom mob

**Comment by RunAwayDSP** [comment]
Unm if thats your code uhh groupzoneid needs to be and actual zone and groupid is the id of the mob in that zone

**Comment by Dev-Roland** [comment]
> Unm if thats your code uhh groupzoneid needs to be and actual zone and groupid is the id of the mob in that zone

I wasn't given an actual example from @Hinekri though he believes that this sample should suffice for this issue report.

**Comment by Hinekri** [comment]
We have many event monsters created thru the local mob = zone:insertDynamicEntity({ objtype = xi.objType.MOB,

 It is not a specific or single one doing the windower crash to him but all of them thus I only dropped an example of how they are built and not just X monster so nobody gets hung up thinking its just the single one.

Should note those of us on Ashita have no issues.

**Comment by zach2good** [comment]
We can't look into an issue unless you give us an actual working/broken example, this doesn't do anything:
```lua

function onTrigger(player)
    local zone = player:getZone()
    local mob = zone:insertDynamicEntity({
        objtype = xi.objType.MOB,
        name = "Name Here",
        x = -6,
        y = -6,
        z = 6,
        rotation = 0,
        groupId = 69,
        groupZoneId = 666,

        onMobSpawn = function(mobArg)
            SetServerVariable("Variable_Here", 0)
        end,

        onMobDeath = function(mobArg, playerArg)
            playerArg:setCharVar("Variable_Here", 1)
        end,

        releaseIdOnDeath = true,
    })

    mob:spawn()

    player:PrintToPlayer(string.format("Spawning (Lv: %i, HP: %i)\n%s", mob:getMainLvl(), mob:getMaxHP(), mob))
end
```

**Comment by zach2good** [comment]
I can also spawn Fafnir with `!fafnir` and cast spells on it in Windower, without a crash:
![image](https://user-images.githubusercontent.com/1389729/173895610-094b08e9-51df-4e47-a0de-17cfea29b26a.png)

**Comment by zach2good** [comment]
lol, wait, this is Ashita

**Comment by zach2good** [comment]
Got this when casting on `!fafnir`:
![image](https://user-images.githubusercontent.com/1389729/173897661-6fd480f2-4da5-4076-8a75-cdc2099f8a59.png)

**Comment by Hinekri** [comment]
local mob = zone:insertDynamicEntity({
                objtype = xi.objType.MOB,
                name = "Test Dummy",
                x = -417.141,
                y = -70.298,
                z = 409.665,
                rotation = 50,
                groupId = 78,
                groupZoneId = 289,

            onMobSpawn = function(mob)
            SetServerVariable("[TestDummy]", 0)
            end,	
    		
            onMobDeath = function(mob, player, isKiller, noKiller)
			    player:setCharVar("[DummyKiller]",1)
            end,
			releaseIdOnDeath = true,
        })
        mob:setDropID(6666)
        mob:setSpawn(-417.141, -70.298, 409.665, 50)
        mob:spawn()
		mob:setMod(xi.mod.STR, 40)
        mob:setMod(xi.mod.VIT, 20)
        mob:setMod(xi.mod.INT, 65)
        mob:setMod(xi.mod.MND, 20)
        mob:setMod(xi.mod.CHR, 20)
        mob:setMod(xi.mod.AGI, 20)
        mob:setMod(xi.mod.DEX, 40)
        mob:setMod(xi.mod.DEFP, 0)
        mob:setMod(xi.mod.RATTP, 0)
        mob:setMod(xi.mod.DEFP, 525)
        mob:setMod(xi.mod.RATTP, 525)
        mob:setMod(xi.mod.ACC, 100)
        mob:setMod(xi.mod.EARTH_RES, 200)
        mob:setMod(xi.mod.DARK_RES, 250)
        mob:setMod(xi.mod.LIGHT_RES, 200)
        mob:setMod(xi.mod.FIRE_RES, 128)
        mob:setMod(xi.mod.WATER_RES, 200)
        mob:setMod(xi.mod.THUNDER_RES, 200)
        mob:setMod(xi.mod.WIND_RES, 200)
        mob:setMod(xi.mod.ICE_RES, 200)
        mob:setMod(xi.mod.SILENCERES, 100)
        mob:setMod(xi.mod.ICE_ABSORB, 100)
        mob:setMod(xi.mod.STUNRES, 50)
        mob:setMod(xi.mod.BINDRES, 100)
        mob:setMod(xi.mod.GRAVITYRES, 100)
        mob:setMod(xi.mod.SLEEPRES, 100)
        mob:setMod(xi.mod.POISONRES, 100)
        mob:setMod(xi.mod.PARALYZERES, 100)
        mob:setMod(xi.mod.LULLABYRES, 100)
        mob:setMod(xi.mod.FASTCAST, 10)
        mob:addStatusEffect(xi.effect.REGEN, 30, 3, 0)
        mob:addStatusEffect(xi.effect.REFRESH, 50, 3, 0)

**Comment by zach2good** [comment]
Alrighty, I found the issue, here's some quotes from the Windower Discord:
```
zach2good — Today at 22:50
When I unload all my addons, I don't crash.
I'm trying to pinpoint the source, but I don't have any spicy custom addons, only popular battletested ones

Nifim — Today at 22:52
popular battledtested ones are really only solid on retail, private server your mileage may vary
timers crashes people on private servers

zach2good — Today at 22:52
Found it, it's Timers
Just repro'd as you mentioned it 🙂

zach2good — Today at 22:53
Is there any way I can poke around inside and find the problem, so I can fix things on my end?

Rubenator — Today at 22:54
Inside Timers? No 

CHA CHA CHIA™ — Today at 22:55
Sorry no but an addon replacement is actively be worked on ATM.
```

Nothing we can do about this if we can't poke around inside Timers, I'm tracking other windower issues here: https://github.com/LandSandBoat/server/issues/2019

**Comment by Dev-Roland** [comment]
I added a related issue to this: these mobs don't generate any battle chat when the Battlemod addon is loaded. I thought it might provide a good symptom to troubleshoot since A) Battlemod actually lets you see all of its code so that you can troubleshoot and B) perhaps the Battlemod and Timers issues are caused by the same root cause. Might get 2 birds with 1 stone.

**Comment by WinterSolstice8** [comment]
That should also be fixed by #2031, it's the same root cause.

**Comment by Dev-Roland** [comment]
Sweet, good job!
